lib/membrane_vpx/encoder/vpx_encoder.ex

```diff
@@ -42,25 +44,23 @@ defmodule Membrane.VPx.Encoder do
 
   @spec handle_stream_format(:input, RawVideo.t(), CallbackContext.t(), State.t()) ::
           callback_return()
-  def handle_stream_format(
-        :input,
-        raw_video_format,
-        _ctx,
-        %State{codec_module: codec_module} = state
-      ) do
+  def handle_stream_format(:input, stream_format, ctx, state) do
     %RawVideo{
       width: width,
       height: height,
-      framerate: framerate,
-      pixel_format: pixel_format
-    } = raw_video_format
+      framerate: framerate
+    } = stream_format
 
     output_stream_format =
-      struct(codec_module, width: width, height: height, framerate: framerate)
+      struct(state.codec_module, width: width, height: height, framerate: framerate)
 
-    native = Native.create!(state.codec, width, height, pixel_format, state.encoding_deadline)
+    {flushed_buffers, encoder_ref} =
+      maybe_recreate_encoder(ctx.pads.input.stream_format, stream_format, state)
```

Comment:
Are you sure that it is the old stream format that is available in the `ctx`?

Reply:
is this field not the last received stream format on given pad? I checked on the call to handle_stream_format and it was set to nil

Reply:
https://hexdocs.pm/membrane_core/Membrane.Element.PadData.html#t:t/0 docs would suggest so too

Reply:
Indeed, but I wasn't sure whether the most recent stream format received on this pad is the one that has just been received and is handled by this callback, or the previous one

Reply:
if it's nil on first call i think it's safe to assume that it's the previous one